src/upload.test.js

```diff
@@ -1,97 +1,155 @@
 import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
 
 import { SkynetClient, defaultSkynetPortalUrl } from "./index";
+import { compareFormData } from "./test_utils.js";
 
-jest.mock("axios");
+const mock = new MockAdapter(axios);
 
 const portalUrl = defaultSkynetPortalUrl;
 const client = new SkynetClient(portalUrl);
 const skylink = "XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg";
 
 describe("uploadFile", () => {
+  const url = `${portalUrl}/skynet/skyfile`;
   const filename = "bar.txt";
   const file = new File(["foo"], filename, {
     type: "text/plain",
   });
 
   beforeEach(() => {
-    axios.post.mockResolvedValue({ data: { skylink } });
+    mock.onPost(url).reply(200, { skylink: skylink });
   });
 
-  it("should send post request with FormData", () => {
-    client.upload(file, {});
+  it("should send formdata with file", async () => {
+    const data = await client.upload(file);
+
+    expect(mock.history.post.length).toBe(1);
+    const request = mock.history.post[0];
+    mock.resetHistory();
+
+    await compareFormData(request.data, [["file", "foo", filename]]);
 
-    expect(axios.post).toHaveBeenCalledWith(
-      `${portalUrl}/skynet/skyfile`,
-      expect.any(FormData), // TODO: Inspect data contents.
-      undefined
-    );
+    expect(data).toEqual({ skylink });
   });
 
-  it("should send register onUploadProgress callback if defined", () => {
+  it("should send register onUploadProgress callback if defined", async () => {
     const newPortal = "https://my-portal.net";
+    const url = `${newPortal}/skynet/skyfile`;
     const client = new SkynetClient(newPortal);
-    client.upload(file, { onUploadProgress: jest.fn() });
-
-    expect(axios.post).toHaveBeenCalledWith(
-      `${newPortal}/skynet/skyfile`,
-      expect.any(FormData), // TODO: Inspect data contents.
-      {
-        onUploadProgress: expect.any(Function),
-      }
-    );
+
+    // Use replyOnce to catch a single request with the new URL.
+    mock.onPost(url).replyOnce(200, { skylink: skylink });
+
+    const data = await client.upload(file, { onUploadProgress: jest.fn() });
+
+    expect(mock.history.post.length).toBe(1);
+    const request = mock.history.post[0];
+    mock.resetHistory();
+
+    expect(request.onUploadProgress).toEqual(expect.any(Function));
```

Comment:
expect the function from first line

Reply:
It means the test detected a real issue then.

Reply:
Oh, it's because we modify `onUploadProgress` in `executeRequest`. Honestly it's enough to check here that `request.onUploadProgress` is defined and equal to any function.